public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -77,20 +136,52 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 			$container = $inside.length > 0 ? $inside.first() : $container;
 		}
 
-		var i = _orderedRecipes.indexOf(recipe) + 1,
-			length = _orderedRecipes.length,
-			testRecipe;
-		for (i; i < length; i++) {
-			testRecipe = _orderedRecipes[i];
+		var additionType = '' + (parentRecipe ? parentRecipe._type : 'BACKBONE') + ' > ' + recipe._type;
 
-			if (testRecipe.isInstantiated() && testRecipe.getViewInstance().$el.parent()[0] == $container[0]) {
-				testRecipe.getViewInstance().$el.before(recipe.getViewInstance().$el);
-				return;
-			}
-		}
+		switch (additionType) {
+			case 'BACKBONE > BACKBONE':
+				var i = _orderedRecipes.indexOf(recipe) + 1,
+				length = _orderedRecipes.length,
+				testRecipe;
+
+				for (i; i < length; i++) {
+					testRecipe = _orderedRecipes[i];
+
+					// If any other views have the same parent, add this element before
+					// those elements in the container element
 
-		// otherwise add on top
-		$container.append(recipe.getViewInstance().$el);
+					if (testRecipe.isInstantiated() && testRecipe.isMounted() && testRecipe.getRootEl().parent()[0] == $container[0]) {
+						testRecipe.getRootEl().before(recipe.getRootEl());
+						return;
+					}
+				}
+				$container.append(recipe.getRootEl());
+				break;
+
+			case 'BACKBONE > REACT':
+				var $proxy = $(document.createElement('div'));
+				$container.append($proxy);
+				ReactDOM.render(recipe._viewInstance, $proxy[0]);
+
+				break;
+
+			case 'REACT > REACT':
```

Comment:
So this is a function that handles adding a recipe to a parent recipe.  Previously this would just add one backbone view el to a parent view's el.

The reason for the addition of the switch statement here is that depending on the type (either `backbone` or `react` right now) the way you add the two recipes together changes.

So adding a backbone view to a parent backbone view: `BACKBONE > BACKBONE` is going to be different than adding a react recipe to a backbone recipe `BACKBONE > REACT` or a backbone element to a react parent `REACT > BACKBONE` (this one not implemented yet though).